lms/static/scripts/frontend_apps/components/ErrorModal.tsx

```diff
@@ -0,0 +1,106 @@
+import { LabeledButton, Modal } from '@hypothesis/frontend-shared';
+import type { ModalProps } from '@hypothesis/frontend-shared/lib/components/Modal';
+import type { ComponentChildren } from 'preact';
+import { useRef } from 'preact/hooks';
+
+import type { ErrorLike } from '../errors';
+import ErrorDisplay from './ErrorDisplay';
+
+type ErrorModalBaseProps = {
+  /**
+   * When `true`, disables the retry button. Only relevant when `onRetry` is
+   * provided.
+   */
+  busy?: boolean;
+  children?: ComponentChildren;
+  /** Text displayed on the Modal's cancel button */
+  cancelLabel?: string;
+
+  /**
+   * A brief contextual description of the error state, which will be passed on
+   * to `ErrorDisplay`. Ignored if `error` is not present.
+   */
+  description?: string;
+
+  /**
+   * When provided, `ErrorDetails` will be rendered, in addition to any
+   * `children`
+   */
+  error?: ErrorLike | null;
+
+  /**
+   * A callback for retrying the failed action. When present, a retry button
+   * will be rendered.
+   */
+  onRetry?: () => void;
+  /** Text for retry button, if present. */
+  retryLabel?: string;
+
+  /**
+   * A callback for closing/canceling the modal. If not provided, the modal
+   * will not be closeable.
+   */
+  onCancel?: () => void;
+  title?: string;
+};
+
+export type ErrorModalProps = Omit<
+  ModalProps,
+  'cancelLabel' | 'children' | 'onCancel' | 'title'
```

Comment:
I presume these props are different in `ModalProps` compared to `ErrorModalBaseProps` eg. in optionality?

Reply:
Presumably :). I'm about to switch this component over to the updated Modal, which has cleaner/updated API, so this is going to change.